daemon/api.go

```diff
@@ -1168,3 +1177,76 @@ func assertsFindMany(c *Command, r *http.Request) Response {
 func getEvents(c *Command, r *http.Request) Response {
 	return EventResponse(c.d.hub)
 }
+
+type changeInfo struct {
+	Kind    string      `json:"kind"`
+	Summary string      `json:"summary"`
+	Status  string      `json:"status"`
+	Tasks   []*taskInfo `json:"tasks,omitempty"`
+}
+
+type taskInfo struct {
+	Kind    string   `json:"kind"`
+	Summary string   `json:"summary"`
+	Status  string   `json:"status"`
+	Log     []string `json:"log,omitempty"`
+}
+
+func getChanges(c *Command, r *http.Request) Response {
+	query := r.URL.Query()
+	logTail := 1
+	qLogTail := query.Get("log-tail")
+	if qLogTail != "" {
+		n, err := strconv.Atoi(qLogTail)
+		if err != nil || n < 0 {
+			return BadRequest("invalid log-tail numeric value")
+		}
+		logTail = n
+	}
+	// XXX: default this to "Done"?
+	excludeStatuses := strings.Split(query.Get("exclude-statuses"), ",")
+	for i, v := range excludeStatuses {
+		excludeStatuses[i] = strings.ToLower(v)
+	}
+	sort.Strings(excludeStatuses)
+	exclude := func(status string) bool {
+		s := strings.ToLower(status)
+		i := sort.SearchStrings(excludeStatuses, s)
+		return i < len(excludeStatuses) && s == excludeStatuses[i]
+	}
+
+	state := c.d.overlord.StateEngine().State()
```

Comment:
We're doing this somewhat often inside the daemon. We should add an Overlord.State method.